snapcraft/plugins/gulp.py

```diff
@@ -99,6 +99,14 @@ def build(self):
         env = os.environ.copy()
         env['PATH'] = '{}:{}'.format(
             os.path.join(self._npm_dir, 'bin'), env['PATH'])
+
+        # make npm use http proxy to access registry via https
+        # ref. https://github.com/npm/npm/issues/2050
+        if 'http_proxy' in os.environ:
+            logger.info('Set http_proxy environment variable to "https-proxy"')
+            self.run(['npm', 'config', '-g', 'set', 'https-proxy',
```

Comment:
Oh, I didn't know that, sorry.

It seems that "`npm_config_https-proxy=...`" env is more useful and safely. I  will change to use it.